src/biblebot/update_check.py

```
@@ -0,0 +1,141 @@
+    Args:
+        current (str): Current version string
+        latest (str): Latest available version string
+
+    Returns:
+        bool: True if latest version is newer than current version
+    """
+    try:
+        current_ver = version.parse(current)
+        latest_ver = version.parse(latest)
+        return latest_ver > current_ver
+    except Exception as e:
```

Ruff (0.12.2)

Do not catch blind exception: `Exception`

(BLE001)